src/compiler/checker.ts

```diff
@@ -5451,6 +5474,18 @@ namespace ts {
             target = getErasedSignature(target);
 
             let result = Ternary.True;
+            if (source.thisType && target.thisType && source.thisType !== voidType) {
+                // void sources are assignable to anything.
+                const related = compareTypes(source.thisType, target.thisType, /*reportErrors*/ false)
+                    || compareTypes(target.thisType, source.thisType, reportErrors);
+                if (!related) {
+                    if (reportErrors) {
+                        errorReporter(Diagnostics.Types_of_parameters_0_and_1_are_incompatible, "this", "this");

```

Comment:
1. "The specified 'this' types of each signature are incompatible.
2. "The 'this'-context for each signature are incompatible."
3. "Types of each 'this' parameter are incompatible

I'm leaning towards 1.